New requirement dropped at 0,0 instead of mouse click

### Describe the bug
After importing and formatting requirements (unclear how relevant this is), I go to drop a new requirement but the position defaults to 0,0.

### To Reproduce
Steps to reproduce the behavior:
1. In SysML
2. Import set of requirements into new model (based on documentation example)
3. Contain requirements up to 3-4 levels deep
4. Auto-format diagram
5. Select Requirement from left side pane
6. Click on diagram to drop requirement

If possible, add the error report from `≡` (top-right menu) → `Tools` → `Error Reports`.

### Additional information
Gaphor version:         3.1.0
Operating System:       Windows (10, AMD64)
Display:                GdkWin32Display
Python version:         3.12.10
GTK version:            4.18.5
Adwaita version:        1.7.4
GtkSourceView version:  5.16.0
Cairo version:          1.18.4
Pango version:          1.56.3
PyGObject version:      3.52.3
Pycairo version:        1.28.0


Errors:

Time since application startup: 0:00:26
|Traceback (most recent call last):
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\diagram\tools\placement.py", line 40, in on_drag_begin
|    item = create_item(
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\diagram\tools\placement.py", line 61, in create_item
|    item = factory(view.model, parent)
|           |       |           -> None
|           |       -> <diagramview.DiagramView object at 0x000001ef0c142280 (DiagramView at 0x000001ef205ccc70)>
|           -> <function new_item_factory.<locals>.item_factory at 0x000001ef083fdb20>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\diagram\diagramtoolbox.py", line 115, in item_factory
|    config_func(item)
|    |           -> <gaphor.SysML.requirements.requirement.RequirementItem element 47d34520-422d-11f0-a4d9-64d69ae029a4>
|    -> <function namespace_config at 0x000001ef0774bba0>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\UML\toolboxconfig.py", line 21, in namespace_config
|    default_namespace(new_item)
|    |                 -> <gaphor.SysML.requirements.requirement.RequirementItem element 47d34520-422d-11f0-a4d9-64d69ae029a4>
|    -> <function default_namespace at 0x000001ef0774bb00>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\UML\toolboxconfig.py", line 16, in default_namespace
|    new_item.subject.package = owner_package(new_item.diagram)
|    |                          |             -> <gaphor.SysML.requirements.requirement.RequirementItem element 47d34520-422d-11f0-a4d9-64d69ae029a4>
|    |                          -> <function owner_package at 0x000001ef044b5800>
|    -> <gaphor.SysML.requirements.requirement.RequirementItem element 47d34520-422d-11f0-a4d9-64d69ae029a4>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\UML\recipes.py", line 444, in owner_package
|    return owner_of_type(element, Package)
|           |             |        -> <class 'gaphor.UML.uml.Package'>
|           |             -> <gaphor.core.modeling.diagram.Diagram element 71607bef-41a2-11f0-a5b7-64d69ae029a4>
|           -> <function owner_of_type at 0x000001ef044b5760>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\UML\recipes.py", line 440, in owner_of_type
|    return owner_of_type(element.owner, owner_type)
|           |             |              -> <class 'gaphor.UML.uml.Package'>
|           |             -> <gaphor.core.modeling.diagram.Diagram element 71607bef-41a2-11f0-a5b7-64d69ae029a4>
|           -> <function owner_of_type at 0x000001ef044b5760>
|AttributeError: 'Diagram' object has no attribute 'owner'
└─────────────────────────────╌┄┈
Time since application startup: 0:05:20
|Traceback (most recent call last):
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\diagram\tools\placement.py", line 40, in on_drag_begin
|    item = create_item(
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\diagram\tools\placement.py", line 61, in create_item
|    item = factory(view.model, parent)
|           |       |           -> None
|           |       -> <diagramview.DiagramView object at 0x000001ef0c142280 (DiagramView at 0x000001ef205ccc70)>
|           -> <function new_item_factory.<locals>.item_factory at 0x000001ef083fdb20>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\diagram\diagramtoolbox.py", line 115, in item_factory
|    config_func(item)
|    |           -> <gaphor.SysML.requirements.requirement.RequirementItem element f7063988-422d-11f0-b3e4-64d69ae029a4>
|    -> <function namespace_config at 0x000001ef0774bba0>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\UML\toolboxconfig.py", line 21, in namespace_config
|    default_namespace(new_item)
|    |                 -> <gaphor.SysML.requirements.requirement.RequirementItem element f7063988-422d-11f0-b3e4-64d69ae029a4>
|    -> <function default_namespace at 0x000001ef0774bb00>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\UML\toolboxconfig.py", line 16, in default_namespace
|    new_item.subject.package = owner_package(new_item.diagram)
|    |                          |             -> <gaphor.SysML.requirements.requirement.RequirementItem element f7063988-422d-11f0-b3e4-64d69ae029a4>
|    |                          -> <function owner_package at 0x000001ef044b5800>
|    -> <gaphor.SysML.requirements.requirement.RequirementItem element f7063988-422d-11f0-b3e4-64d69ae029a4>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\UML\recipes.py", line 444, in owner_package
|    return owner_of_type(element, Package)
|           |             |        -> <class 'gaphor.UML.uml.Package'>
|           |             -> <gaphor.core.modeling.diagram.Diagram element 71607bef-41a2-11f0-a5b7-64d69ae029a4>
|           -> <function owner_of_type at 0x000001ef044b5760>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\UML\recipes.py", line 440, in owner_of_type
|    return owner_of_type(element.owner, owner_type)
|           |             |              -> <class 'gaphor.UML.uml.Package'>
|           |             -> <gaphor.core.modeling.diagram.Diagram element 71607bef-41a2-11f0-a5b7-64d69ae029a4>
|           -> <function owner_of_type at 0x000001ef044b5760>
|AttributeError: 'Diagram' object has no attribute 'owner'
└─────────────────────────────╌┄┈
Time since application startup: 0:07:05
|Traceback (most recent call last):
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\diagram\tools\placement.py", line 40, in on_drag_begin
|    item = create_item(
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\diagram\tools\placement.py", line 61, in create_item
|    item = factory(view.model, parent)
|           |       |           -> None
|           |       -> <diagramview.DiagramView object at 0x000001ef0c142280 (DiagramView at 0x000001ef205ccc70)>
|           -> <function new_item_factory.<locals>.item_factory at 0x000001ef083fdb20>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\diagram\diagramtoolbox.py", line 115, in item_factory
|    config_func(item)
|    |           -> <gaphor.SysML.requirements.requirement.RequirementItem element 36029378-422e-11f0-9cb5-64d69ae029a4>
|    -> <function namespace_config at 0x000001ef0774bba0>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\UML\toolboxconfig.py", line 21, in namespace_config
|    default_namespace(new_item)
|    |                 -> <gaphor.SysML.requirements.requirement.RequirementItem element 36029378-422e-11f0-9cb5-64d69ae029a4>
|    -> <function default_namespace at 0x000001ef0774bb00>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\UML\toolboxconfig.py", line 16, in default_namespace
|    new_item.subject.package = owner_package(new_item.diagram)
|    |                          |             -> <gaphor.SysML.requirements.requirement.RequirementItem element 36029378-422e-11f0-9cb5-64d69ae029a4>
|    |                          -> <function owner_package at 0x000001ef044b5800>
|    -> <gaphor.SysML.requirements.requirement.RequirementItem element 36029378-422e-11f0-9cb5-64d69ae029a4>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\UML\recipes.py", line 444, in owner_package
|    return owner_of_type(element, Package)
|           |             |        -> <class 'gaphor.UML.uml.Package'>
|           |             -> <gaphor.core.modeling.diagram.Diagram element 71607bef-41a2-11f0-a5b7-64d69ae029a4>
|           -> <function owner_of_type at 0x000001ef044b5760>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\UML\recipes.py", line 440, in owner_of_type
|    return owner_of_type(element.owner, owner_type)
|           |             |              -> <class 'gaphor.UML.uml.Package'>
|           |             -> <gaphor.core.modeling.diagram.Diagram element 71607bef-41a2-11f0-a5b7-64d69ae029a4>
|           -> <function owner_of_type at 0x000001ef044b5760>
|AttributeError: 'Diagram' object has no attribute 'owner'
└─────────────────────────────╌┄┈
Time since application startup: 0:10:47
|Traceback (most recent call last):
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\diagram\tools\placement.py", line 40, in on_drag_begin
|    item = create_item(
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\diagram\tools\placement.py", line 61, in create_item
|    item = factory(view.model, parent)
|           |       |           -> None
|           |       -> <diagramview.DiagramView object at 0x000001ef0c142280 (DiagramView at 0x000001ef205ccc70)>
|           -> <function new_item_factory.<locals>.item_factory at 0x000001ef083fdb20>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\diagram\diagramtoolbox.py", line 115, in item_factory
|    config_func(item)
|    |           -> <gaphor.SysML.requirements.requirement.RequirementItem element ba4206e4-422e-11f0-bfc0-64d69ae029a4>
|    -> <function namespace_config at 0x000001ef0774bba0>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\UML\toolboxconfig.py", line 21, in namespace_config
|    default_namespace(new_item)
|    |                 -> <gaphor.SysML.requirements.requirement.RequirementItem element ba4206e4-422e-11f0-bfc0-64d69ae029a4>
|    -> <function default_namespace at 0x000001ef0774bb00>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\UML\toolboxconfig.py", line 16, in default_namespace
|    new_item.subject.package = owner_package(new_item.diagram)
|    |                          |             -> <gaphor.SysML.requirements.requirement.RequirementItem element ba4206e4-422e-11f0-bfc0-64d69ae029a4>
|    |                          -> <function owner_package at 0x000001ef044b5800>
|    -> <gaphor.SysML.requirements.requirement.RequirementItem element ba4206e4-422e-11f0-bfc0-64d69ae029a4>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\UML\recipes.py", line 444, in owner_package
|    return owner_of_type(element, Package)
|           |             |        -> <class 'gaphor.UML.uml.Package'>
|           |             -> <gaphor.core.modeling.diagram.Diagram element 71607bef-41a2-11f0-a5b7-64d69ae029a4>
|           -> <function owner_of_type at 0x000001ef044b5760>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\UML\recipes.py", line 440, in owner_of_type
|    return owner_of_type(element.owner, owner_type)
|           |             |              -> <class 'gaphor.UML.uml.Package'>
|           |             -> <gaphor.core.modeling.diagram.Diagram element 71607bef-41a2-11f0-a5b7-64d69ae029a4>
|           -> <function owner_of_type at 0x000001ef044b5760>
|AttributeError: 'Diagram' object has no attribute 'owner'
└─────────────────────────────╌┄┈
Time since application startup: 0:11:16
|Traceback (most recent call last):
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\SysML\propertypages.py", line 110, in _on_show_text_change
|    with Transaction(self.event_manager):
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\transaction.py", line 58, in __init__
|    self._handle(TransactionBegin(self.context))
|    |            |                -> <gaphor.transaction.Transaction object at 0x000001ef110b7080>
|    |            -> <class 'gaphor.event.TransactionBegin'>
|    -> <gaphor.transaction.Transaction object at 0x000001ef110b7080>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\transaction.py", line 113, in _handle
|    self.event_manager.handle(event)
|    |                         -> TransactionBegin(context=None)
|    -> <gaphor.transaction.Transaction object at 0x000001ef110b7080>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\core\eventmanager.py", line 126, in handle
|    self._priority.handle(event)
|    |                     -> TransactionBegin(context=None)
|    -> <gaphor.core.eventmanager.EventManager object at 0x000001ef08486db0>
|  File "C:\msys64\home\hbartsch\gaphor\.venv\lib\python3.12\site-packages\generic\event.py", line 63, in handle
|    raise ExceptionGroup("Error while handling events", exceptions)
|                                                        -> [AssertionError('assert not self._current_transaction')]
|ExceptionGroup: Error while handling events (1 sub-exception)
└─┬──────────────────────────────╌┄┈
  |Traceback (most recent call last):
  |  File "C:\msys64\home\hbartsch\gaphor\.venv\lib\python3.12\site-packages\generic\event.py", line 59, in handle
  |    handler(event)
  |    |       -> TransactionBegin(context=None)
  |    -> <bound method UndoManager.begin_transaction of <gaphor.services.undomanager.UndoManager object at 0x000001ef084940b0>>
  |  File "C:\msys64\home\hbartsch\gaphor\gaphor\services\undomanager.py", line 204, in begin_transaction
  |    assert not self._current_transaction
  |               -> <gaphor.services.undomanager.UndoManager object at 0x000001ef084940b0>
  |AssertionError: assert not self._current_transaction
  └─────────────────────────────╌┄┈
Time since application startup: 0:11:25
|Traceback (most recent call last):
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\SysML\propertypages.py", line 110, in _on_show_text_change
|    with Transaction(self.event_manager):
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\transaction.py", line 58, in __init__
|    self._handle(TransactionBegin(self.context))
|    |            |                -> <gaphor.transaction.Transaction object at 0x000001ef1115f9b0>
|    |            -> <class 'gaphor.event.TransactionBegin'>
|    -> <gaphor.transaction.Transaction object at 0x000001ef1115f9b0>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\transaction.py", line 113, in _handle
|    self.event_manager.handle(event)
|    |                         -> TransactionBegin(context=None)
|    -> <gaphor.transaction.Transaction object at 0x000001ef1115f9b0>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\core\eventmanager.py", line 126, in handle
|    self._priority.handle(event)
|    |                     -> TransactionBegin(context=None)
|    -> <gaphor.core.eventmanager.EventManager object at 0x000001ef08486db0>
|  File "C:\msys64\home\hbartsch\gaphor\.venv\lib\python3.12\site-packages\generic\event.py", line 63, in handle
|    raise ExceptionGroup("Error while handling events", exceptions)
|                                                        -> [AssertionError('assert not self._current_transaction')]
|ExceptionGroup: Error while handling events (1 sub-exception)
└─┬──────────────────────────────╌┄┈
  |Traceback (most recent call last):
  |  File "C:\msys64\home\hbartsch\gaphor\.venv\lib\python3.12\site-packages\generic\event.py", line 59, in handle
  |    handler(event)
  |    |       -> TransactionBegin(context=None)
  |    -> <bound method UndoManager.begin_transaction of <gaphor.services.undomanager.UndoManager object at 0x000001ef084940b0>>
  |  File "C:\msys64\home\hbartsch\gaphor\gaphor\services\undomanager.py", line 204, in begin_transaction
  |    assert not self._current_transaction
  |               -> <gaphor.services.undomanager.UndoManager object at 0x000001ef084940b0>
  |AssertionError: assert not self._current_transaction
  └─────────────────────────────╌┄┈
Time since application startup: 0:11:28
|Traceback (most recent call last):
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\SysML\propertypages.py", line 110, in _on_show_text_change
|    with Transaction(self.event_manager):
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\transaction.py", line 58, in __init__
|    self._handle(TransactionBegin(self.context))
|    |            |                -> <gaphor.transaction.Transaction object at 0x000001ef185e50a0>
|    |            -> <class 'gaphor.event.TransactionBegin'>
|    -> <gaphor.transaction.Transaction object at 0x000001ef185e50a0>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\transaction.py", line 113, in _handle
|    self.event_manager.handle(event)
|    |                         -> TransactionBegin(context=None)
|    -> <gaphor.transaction.Transaction object at 0x000001ef185e50a0>
|  File "C:\msys64\home\hbartsch\gaphor\gaphor\core\eventmanager.py", line 126, in handle
|    self._priority.handle(event)
|    |                     -> TransactionBegin(context=None)
|    -> <gaphor.core.eventmanager.EventManager object at 0x000001ef08486db0>
|  File "C:\msys64\home\hbartsch\gaphor\.venv\lib\python3.12\site-packages\generic\event.py", line 63, in handle
|    raise ExceptionGroup("Error while handling events", exceptions)
|                                                        -> [AssertionError()]
|ExceptionGroup: Error while handling events (1 sub-exception)
└─┬──────────────────────────────╌┄┈
  |Traceback (most recent call last):
  |  File "C:\msys64\home\hbartsch\gaphor\.venv\lib\python3.12\site-packages\generic\event.py", line 59, in handle
  |    handler(event)
  |    |       -> TransactionBegin(context=None)
  |    -> <bound method UndoManager.begin_transaction of <gaphor.services.undomanager.UndoManager object at 0x000001ef084940b0>>
  |  File "C:\msys64\home\hbartsch\gaphor\gaphor\services\undomanager.py", line 204, in begin_transaction
  |    assert not self._current_transaction
  |               -> <gaphor.services.undomanager.UndoManager object at 0x000001ef084940b0>
  |AssertionError: assert not self._current_transaction
  └─────────────────────────────╌┄┈

--- a/gaphor/UML/actions/actionstoolbox.py
+++ b/gaphor/UML/actions/actionstoolbox.py
@@ -6,6 +6,7 @@ from gaphas.item import SE
 
 from gaphor import UML
 from gaphor.diagram.diagramtoolbox import ToolDef, ToolSection, new_item_factory
+from gaphor.diagram.group import owner_of_type
 from gaphor.i18n import gettext, i18nize
 from gaphor.UML import diagramitems
 from gaphor.UML.recipes import owner_package
@@ -21,7 +22,7 @@ def activity_config(new_item, name=None):
 
     diagram = new_item.diagram
 
-    if owner_activity := UML.recipes.owner_of_type(diagram, UML.Activity):
+    if owner_activity := owner_of_type(diagram, UML.Activity):
         subject.activity = owner_activity
         return
 

--- a/gaphor/UML/recipes.py
+++ b/gaphor/UML/recipes.py
@@ -10,9 +10,9 @@ from __future__ import annotations
 
 import itertools
 from collections.abc import Iterable, Sequence
-from typing import TypeVar
 
 from gaphor.core.modeling import UnlimitedNatural
+from gaphor.diagram.group import owner_of_type
 from gaphor.UML.uml import (
     Artifact,
     Association,
@@ -50,8 +50,6 @@ from gaphor.UML.uml import (
     Usage,
     ValueSpecification,
 )
-
-T = TypeVar("T", bound=Element)
 
 
 def stereotypes_str(element: Element, stereotypes: Sequence[str] = ()) -> str:
@@ -432,12 +430,6 @@ def clone_message(msg, inverted=False):
         message.sendEvent = send
         message.receiveEvent = receive
     return message
-
-
-def owner_of_type(element: Element | None, owner_type: type[T]) -> T | None:
-    if element is None or isinstance(element, owner_type):
-        return element
-    return owner_of_type(element.owner, owner_type)
 
 
 def owner_package(element: Element | None) -> Package | None:

--- a/gaphor/diagram/group.py
+++ b/gaphor/diagram/group.py
@@ -10,10 +10,13 @@ import itertools
 from collections.abc import Callable, Iterator
 from enum import Enum
 from functools import singledispatch
+from typing import TypeVar
 
 from generic.multidispatch import FunctionDispatcher, multidispatch
 
 from gaphor.core.modeling import Base, Diagram, Presentation
+
+T = TypeVar("T", bound=Base)
 
 
 def self_and_owners(element: Base | RootType | None) -> Iterator[Base]:
@@ -26,6 +29,15 @@ def self_and_owners(element: Base | RootType | None) -> Iterator[Base]:
         yield e
         seen.add(e)
         e = owner(e)
+
+
+def owner_of_type(element: Base | None, owner_type: type[T]) -> T | None:
+    if element is None or isinstance(element, owner_type):
+        return element
+
+    if (o := owner(element)) is Root:
+        return None
+    return owner_of_type(o, owner_type)
 
 
 def change_owner(new_parent: Base | None, element: Base) -> bool:

